ironfish/src/wallet/scanner/walletScanner.ts

```diff
@@ -177,34 +182,32 @@ export class WalletScanner {
       )
     }
 
-    const connectOnlyAccounts = new Array<Account>()
-    const decryptAndConnectAccounts = new Array<Account>()
-
     for (const candidate of this.scanningAccounts) {
-      if (
-        !candidate.scanFrom ||
-        BufferUtils.equalsNullable(candidate.scanFrom.hash, blockHeader.previousBlockHash)
-      ) {
-        candidate.scanFrom = null
-
-        if (
-          candidate.account.createdAt === null ||
-          blockHeader.sequence >= candidate.account.createdAt.sequence
-        ) {
-          decryptAndConnectAccounts.push(candidate.account)
-        } else {
-          connectOnlyAccounts.push(candidate.account)
+      const { account, scanFrom } = candidate
+
+      if (scanFrom === 'always-scan') {
+        continue
+      }
+
+      if (scanFrom.head === null) {
+        if (account.createdAt === null && blockHeader.sequence === GENESIS_BLOCK_SEQUENCE) {
+          candidate.scanFrom = 'always-scan'
+        }
+
+        if (account.createdAt && blockHeader.sequence >= account.createdAt.sequence) {
+          candidate.scanFrom = 'always-scan'
         }
       }
-    }
 
-    for (const account of connectOnlyAccounts) {
-      if (abort?.signal.aborted) {
-        return
+      if (scanFrom.head?.hash.equals(blockHeader.previousBlockHash)) {
+        candidate.scanFrom = 'always-scan'
       }
-      await this.wallet.connectBlockForAccount(account, blockHeader, [], false)
     }
 
+    const decryptAndConnectAccounts = this.scanningAccounts
+      .filter(({ scanFrom }) => scanFrom === 'always-scan')
+      .map(({ account }) => account)
+
```

Comment:
It seems like we use `'always-scan'` as a way to to divide `scanningAccounts` into two lists: one list that the walletScanner is currently scanning for (the `always-scan` accounts), and a list of accounts that are waiting for the walletScanner to get to them.

I wonder if it would be easier to keep the accounts separate or if it would make it harder to follow?

Reply:
Might be easier but popping and pushing the accounts across arrays would be more verbose. I can look into something like this though